---
title: "WQS Test of Selection"
author: "Timothy M. Beissinger"
date: "18 May, 2017"
output: html_document
---

Data from Lorenz, 2015 (G3) will be re-visited to test the hypothesis that the reason selection mapping and GWAS yielded no results, despite moderate to high heritabilities and phenotypic change, was because the selected traits were highly quantitative. To do this, allelic effects estimates from GWAS will be imported and allele frequency change from cycle 2 to cycle 5 will be computed. Then, the sum of frequency change multiplied by effect size will be computed. If GWAS effects and frequency changes are related, this sum should be non-zero and in the direction that the trait was selected. A permutation test will be conducted to test for significance.

### Data loading and prep

First we read in the Genotypes and phenotypes. There are not many markers here, and the reason why not is not explained in the paper... Something to ask Aaron about.

```{r}
### genos
setwd("/home/beissinger/Documents/ComplexSelection/Maize")
wqsGeno <- read.csv('../../WQS/DATA/genoWQS_randImp.csv',header=T,stringsAsFactors=F)
```

Now we'll import the map.

```{r}
wqsMap <- read.csv('../../WQS/DATA/wqsMap_randImp.csv',stringsAsFactors=F)
```

Gapit cannot work with non-integer chromosomes. Remove those labeled "0"
```{r}
remMap <- which(wqsMap$Chromosome == 0)
remGeno <- remMap+1

wqsGeno<-wqsGeno[,-remGeno]
wqsMap<-wqsMap[-remMap,]
```

Now split individuals into different data frames according to their cycle

```{r}
names(wqsGeno)[1] <- "Taxa"
CycleIndicator <- as.numeric(unlist(strsplit(wqsGeno$Taxa,split="_C"))[seq(2,2*nrow(wqsGeno),2)])

Cycle0 <- wqsGeno[which(CycleIndicator == 0),]
Cycle1 <- wqsGeno[which(CycleIndicator == 1),]
Cycle2 <- wqsGeno[which(CycleIndicator == 2),]
Cycle3 <- wqsGeno[which(CycleIndicator == 3),]
Cycle4 <- wqsGeno[which(CycleIndicator == 4),]
Cycle5 <- wqsGeno[which(CycleIndicator == 5),]
CycleList <- list(Cycle0,Cycle1,Cycle2,Cycle3,Cycle4,Cycle5)
```

### Analysis

The next step is to calculate the allele frequency at each cycle. 

```{r}
frequencies<-matrix(nrow=ncol(wqsGeno)-1,ncol=6)
for(i in 1:6){
frequencies[,i] <- colMeans(CycleList[[i]][,-1],na.rm=T)/2
}
frequencies <- as.data.frame(frequencies)
names(frequencies) <- c("Cycle0","Cycle1","Cycle2","Cycle3","Cycle4","Cycle5")
```

Now compute allele frequency change from cycle 2 to 5 and make a histogram
```{r}
change<-frequencies$Cycle5-frequencies$Cycle2
hist(change,main=paste("Histogram of C2 to C5 Frequency Change, Mean=",round(mean(change,na.rm=T),digits=3),sep=""))
```


Now load in phenotypes to enable calculating effects estimates.
```{r}
### phenos
setwd("/home/beissinger/Documents/WQS_Prediction/GWAS_effects/")
adf <- read.csv('../Previous_G3/wqsSupportingData/adfBlups.csv', head=TRUE,stringsAsFactors=F)
ndf <- read.csv('../Previous_G3/wqsSupportingData/ndfBlups.csv', head=TRUE,stringsAsFactors=F)
ivd <- read.csv('../Previous_G3/wqsSupportingData/ivdBlups.csv', head=TRUE,stringsAsFactors=F)
cp <- read.csv('../Previous_G3/wqsSupportingData/cpBlups.csv', head=TRUE,stringsAsFactors=F)
drymat <- read.csv('../Previous_G3/wqsSupportingData/dmBlups.csv', head=TRUE,stringsAsFactors=F)
starch <- read.csv('../Previous_G3/wqsSupportingData/starchBlups.csv', head=TRUE,stringsAsFactors=F)
yield <- read.csv('../Previous_G3/wqsSupportingData/yieldBlups.csv', head=TRUE,stringsAsFactors=F)

### restore working directory
setwd("/home/beissinger/Documents/ComplexSelection/Maize")
```

Now reformat genotypes file. Reorganize by coding to -1,0,1 and so that markers are columns, individuals are rows.
```{r}
geno<-wqsGeno
geno[,2:nrow(geno)] <- 1-geno[,2:nrow(geno)]
```

Now write a function to compute effects using the rrBLUP package
```{r}
library(rrBLUP)
rrblupFunction <- function(geno, pheno, map,main){
  ##drop "_CX" from rownames in geno
  names(geno)[1] <- "Taxa"
  geno$Taxa <- gsub('...$', '', wqsGeno$Taxa) 

  ##remove inds without geno & pheno
    ndx <- match(geno[, 1], pheno[, 1])
    noPheno <- geno[, 1][which(is.na(ndx))]
    geno2 <- geno[-which(is.na(ndx)), ]
    ndx2 <- ndx[-which(is.na(ndx))]
    pheno2 <- pheno[ndx2, ]
    
    ##Calculate numbers for each cycle
    cycleIndNum <- vector(length=5)
    cycleIndNum[1] <- length(which(pheno2$Cycle=='C0'))
    cycleIndNum[2] <- length(which(pheno2$Cycle=='C1'))
    cycleIndNum[3] <- length(which(pheno2$Cycle=='C2'))
    cycleIndNum[4] <- length(which(pheno2$Cycle=='C3'))
    cycleIndNum[5] <- length(which(pheno2$Cycle=='C4'))
    yr <- as.character(pheno2$Cycle)
    yr <- as.factor(yr)
    pheno2 <- pheno2[, c(1, 2)]
    
    ##Remove individuals with missing phenoypte (these rows were in phenotype file as NA)
    pheno3<-pheno2[which(is.na(pheno2[,2])==F),]
    geno3<-geno2[which(is.na(pheno2[,2])==F),]
    print(paste("In total", nrow(geno3), "individuals."))

    #run rrBLUP

    
    #result <- mixed.solve(pheno2[,2], Z= as.matrix(geno2[,2:ncol(geno2)]),   K=NULL, SE=F, return.Hinv=FALSE,method="ML") # cycle NOT included as a covariate!
    
    #below is code to include cycle as a covariate
    fixedMat <- model.matrix(pheno2[,2]~yr) #model matrix for years
    
    result <- mixed.solve(pheno3[,2], Z= as.matrix(geno3[,2:ncol(geno3)]),X= model.matrix(pheno2[,2]~yr),   K=NULL, SE=F, return.Hinv=FALSE,method="ML") # cycle INCLUDED as a covariate!

    ##Plot predicted vs actual
    effects <- result$u
    effects.mat <- as.matrix(effects)
    predicted <- as.matrix(geno2[,2:ncol(geno2)]) %*% effects.mat
    predicted = predicted[,1] + result$beta[1]
    plot(pheno2[,2],predicted,xlab="True phenos",ylab="Predicted phenos",main=main)
    
    return(result$u)
}


```

Now run rrBLUP
```{r}
effects.ADF<-rrblupFunction(geno,adf,wqsMap,main="ADF")
effects.NDF<-rrblupFunction(geno,ndf,wqsMap,main="NDF")
effects.IVD<-rrblupFunction(geno,ivd,wqsMap,main="IVD")
effects.CP<-rrblupFunction(geno,cp,wqsMap,main="Crude Protein")
effects.drymat<-rrblupFunction(geno,drymat,wqsMap,main="Dry matter")
effects.starch<-rrblupFunction(geno,starch,wqsMap,main="Starch")
effects.yield<-rrblupFunction(geno,yield,wqsMap,main="Yield")
```

Now apply function to unscaled data--OLD METHOD SO THIS SECTION OF CODE IS SKIPPED
```{r, eval=FALSE}

source("/home/beissinger/Documents/ComplexSelection/R-Function/Ghat.R")
result.adf <- Ghat_func_effectsKnown(effects=effects.ADF, change=change, method = "vanilla", perms=1000 ,plot="Both", num_eff = NULL)

result.ndf <- Ghat_func_effectsKnown(effects=effects.NDF, change=change, method = "vanilla", perms=1000 ,plot="Both", num_eff = NULL)

result.ivd <- Ghat_func_effectsKnown(effects=effects.IVD, change=change, method = "vanilla", perms=1000 ,plot="Both", num_eff = NULL)

result.cp <- Ghat_func_effectsKnown(effects=effects.CP, change=change, method = "vanilla", perms=1000 ,plot="Both", num_eff = NULL)

result.drymat <- Ghat_func_effectsKnown(effects=effects.drymat, change=change, method = "vanilla", perms=1000 ,plot="Both", num_eff = NULL)

result.starch <- Ghat_func_effectsKnown(effects=effects.starch, change=change, method = "vanilla", perms=1000 ,plot="Both", num_eff = NULL)

result.yield <- Ghat_func_effectsKnown(effects=effects.yield, change=change, method = "vanilla", perms=1000 ,plot="Both", num_eff = NULL)

```


Now sum effective number of markers per chromosome. LD Decay calculated in a separate script. Here I'm using number of markers until R^2 <= 0.03.
```{r}
#decay <- c(111,122,129,147,96,72,111,114,91,226) # 0.05
decay <- c(163,234,182,239,178,115,172,189,141,337)
effective.marker.num <- sum(table(wqsMap$Chromosome)/decay)
effective.marker.num
```

Now apply with appropriate scaling to obtain actual results
```{r}
source("/home/beissinger/Documents/ComplexSelection/R-Function/Ghat.R")
result.adf <- Ghat_func_effectsKnown(effects=effects.ADF, change=change, method = "scale", perms=1000 ,plot="Ghat", num_eff = effective.marker.num)
mtext(paste("WQS ADF test for selection, pval = ", round(result.adf$p.val,4)))
result.adf$p.val

result.ndf <- Ghat_func_effectsKnown(effects=effects.NDF, change=change, method = "scale", perms=1000 ,plot="Ghat", num_eff = effective.marker.num)
mtext(paste("WQS NDF test for selection, pval = ", round(result.ndf$p.val,4)))
result.ndf$p.val

result.ivd <- Ghat_func_effectsKnown(effects=effects.IVD, change=change, method = "scale", perms=1000 ,plot="Ghat", num_eff = effective.marker.num)
mtext(paste("WQS IVD test for selection, pval = ", round(result.ivd$p.val,4)))
result.ivd$p.val

result.cp <- Ghat_func_effectsKnown(effects=effects.CP, change=change, method = "scale", perms=1000 ,plot="Ghat", num_eff = effective.marker.num)
mtext(paste("WQS CP test for selection, pval = ", round(result.cp$p.val,4)))
result.cp$p.val

result.drymat <- Ghat_func_effectsKnown(effects=effects.drymat, change=change, method = "scale", perms=1000 ,plot="Ghat", num_eff = effective.marker.num)
mtext(paste("WQS drymatter test for selection, pval = ", round(result.drymat$p.val,4)))
result.drymat$p.val

result.starch <- Ghat_func_effectsKnown(effects=effects.starch, change=change, method = "scale", perms=1000 ,plot="Ghat", num_eff = effective.marker.num)
mtext(paste("WQS starch test for selection, pval = ", round(result.starch$p.val,4)))
result.starch$p.val

result.yield <- Ghat_func_effectsKnown(effects=effects.yield, change=change, method = "scale", perms=1000 ,plot="Ghat", num_eff = effective.marker.num)
mtext(paste("WQS yield test for selection, pval = ", round(result.yield$p.val,4)))
result.yield$p.val
```


Now make a combined plot for all traits, to be used in the manuscript
```{r}
#pdf(file = "maize_manuscript_plot.pdf", height = 8, width = 6)
#png(file = "maize_manuscript_plot.png", height = 900, width = 600)
jpeg(file = "maize_manuscript_plot_reverse_inlay.jpg", height = 8, width = 6, ,units="in",res=300)
par(cex=0.6,mar=c(5,4,2,2)+0.1)
#par(mfrow=c(3,2))
## some pretty colors
colfunc <- colorRampPalette(c("lightblue", "darkred","red","yellow"))
library(MASS)

#adf
par(fig=c(0,0.5,.66666,1))
z_adf <- kde2d(change, effects.ADF, n=200)
plot(change,effects.ADF,xaxt="s",yaxt="s",xlab="Frequency change",ylab="Effect size",main="ADF",cex.axis=0.75)
contour(z_adf,drawlabels=F,nlevels=100,col=colfunc(200),add=T)
abline(lm(effects.ADF~change),col="darkred",lwd=3)

#ivd
par(fig=c(0.5,1,.66666,1),new=T)
z_ivd <- kde2d(change, effects.IVD, n=200)
plot(change,effects.IVD,xaxt="s",yaxt="s",xlab="Frequency change",ylab="Effect size",main="In-vitro digestibility",cex.axis=0.75)
contour(z_ivd,drawlabels=F,nlevels=100,col=colfunc(200),add=T)
abline(lm(effects.IVD~change),col="darkred",lwd=3)

#cp
par(fig=c(0,0.5,.33333,.66666),new=T)
z_CP <- kde2d(change, effects.CP, n=200)
plot(change,effects.CP,xaxt="s",yaxt="s",xlab="Frequency change",ylab="Effect size",main="Protein",cex.axis=0.75)
contour(z_CP,drawlabels=F,nlevels=100,col=colfunc(200),add=T)
abline(lm(effects.CP~change),col="darkred",lwd=3)

#starch
par(fig=c(0.5,1,.33333,.66666),new=T)
z_starch <- kde2d(change, effects.starch, n=200)
plot(change,effects.starch,xaxt="s",yaxt="s",xlab="Frequency change",ylab="Effect size",main="Starch",cex.axis=0.75)
contour(z_starch,drawlabels=F,nlevels=100,col=colfunc(200),add=T)
abline(lm(effects.starch~change),col="darkred",lwd=3)

#yield
par(fig=c(0,0.5,0,.33333),new=T)
z_yield <- kde2d(change, effects.yield, n=200)
plot(change,effects.yield,xaxt="s",yaxt="s",xlab="Frequency change",ylab="Effect size",cex.axis=0.75,main="Yield")
contour(z_yield,drawlabels=F,nlevels=100,col=colfunc(200),add=T)
abline(lm(effects.yield~change),col="darkred",lwd=3)

#drymat
par(fig=c(0.5,1,0,.33333),new=T)
z_drymat <- kde2d(change, effects.drymat, n=200)
plot(change,effects.drymat,xaxt="s",yaxt="s",xlab="Frequency change",ylab="Effect size",cex.axis=0.75,main = "Dry Matter")
contour(z_drymat,drawlabels=F,nlevels=100,col=colfunc(200),add=T)
abline(lm(effects.drymat~change),col="darkred",lwd=3)

#############
###insets####
#############
par(cex=0.55)
#adf
par(fig=c(.33-.04,.49-.04,.85,.965),mgp=c(1,.3,0), new=TRUE)
par(mar=c(2,2,0,0))
left <- min(mean(result.adf$Ghat_perm)-4*result.adf$sd_eff,result.adf$Ghat)
right <- max(mean(result.adf$Ghat_perm)+4*result.adf$sd_eff,result.adf$Ghat)
plot(seq(left,right,length.out=1000),dnorm(seq(left,right,length.out=1000),mean=mean(result.adf$Ghat_perm),sd=result.adf$sd_eff),col="red",lwd=2,type="l",yaxt="n",ylab="",xlab="",xaxt="n",main="")
rect(par("usr")[1], par("usr")[3], par("usr")[2], par("usr")[4], col = "gray")
lines(seq(left,right,length.out=1000),dnorm(seq(left,right,length.out=1000),mean=mean(result.adf$Ghat_perm),sd=result.adf$sd_eff),col="red",lwd=2)
abline(v=result.adf$Ghat,lwd=2,col="darkblue")
text(par("usr")[1]+.8*(par("usr")[2]-par("usr")[1]),par("usr")[3]+0.8*(par("usr")[4]-par("usr")[3]),paste("p =  \n", round(result.adf$p.val,3)),cex=1.1,font=2)


#ivd
par(fig=c(.83-.275,.99-.275,.85,.965),mgp=c(1,.3,0), new=TRUE)
par(mar=c(2,2,0,0))
left <- min(mean(result.ivd$Ghat_perm)-4*result.ivd$sd_eff,result.ivd$Ghat)
right <- max(mean(result.ivd$Ghat_perm)+4*result.ivd$sd_eff,result.ivd$Ghat)
plot(seq(left,right,length.out=1000),dnorm(seq(left,right,length.out=1000),mean=mean(result.ivd$Ghat_perm),sd=result.ivd$sd_eff),col="red",lwd=2,type="l",yaxt="n",ylab="",xlab="",xaxt="n",main="")
rect(par("usr")[1], par("usr")[3], par("usr")[2], par("usr")[4], col = "gray")
lines(seq(left,right,length.out=1000),dnorm(seq(left,right,length.out=1000),mean=mean(result.ivd$Ghat_perm),sd=result.ivd$sd_eff),col="red",lwd=2)
abline(v=result.ivd$Ghat,lwd=2,col="darkblue")
text(par("usr")[1]+.18*(par("usr")[2]-par("usr")[1]),par("usr")[3]+0.8*(par("usr")[4]-par("usr")[3]),paste("p =  \n", round(result.ivd$p.val,3)),cex=1.1,font=2)


#protein
par(fig=c(.33-.04,.49-.04,.85-.33333,.965-.33333),mgp=c(1,.3,0), new=TRUE)
par(mar=c(2,2,0,0))
left <- min(mean(result.cp$Ghat_perm)-4*result.cp$sd_eff,result.cp$Ghat)
right <- max(mean(result.cp$Ghat_perm)+4*result.cp$sd_eff,result.cp$Ghat)
plot(seq(left,right,length.out=1000),dnorm(seq(left,right,length.out=1000),mean=mean(result.cp$Ghat_perm),sd=result.cp$sd_eff),col="red",lwd=2,type="l",yaxt="n",ylab="",xlab="",xaxt="n",main="")
rect(par("usr")[1], par("usr")[3], par("usr")[2], par("usr")[4], col = "gray")
lines(seq(left,right,length.out=1000),dnorm(seq(left,right,length.out=1000),mean=mean(result.cp$Ghat_perm),sd=result.cp$sd_eff),col="red",lwd=2)
abline(v=result.cp$Ghat,lwd=2,col="darkblue")
text(par("usr")[1]+.8*(par("usr")[2]-par("usr")[1]),par("usr")[3]+0.8*(par("usr")[4]-par("usr")[3]),paste("p =  \n", round(result.cp$p.val,3)),cex=1.1,font=2)


#starch
par(fig=c(.83-.275,.99-.275,.85-.33333,.965-.33333),mgp=c(1,.3,0), new=TRUE)
par(mar=c(2,2,0,0))
left <- min(mean(result.starch$Ghat_perm)-4*result.starch$sd_eff,result.starch$Ghat)
right <- max(mean(result.starch$Ghat_perm)+4*result.starch$sd_eff,result.starch$Ghat)
plot(seq(left,right,length.out=1000),dnorm(seq(left,right,length.out=1000),mean=mean(result.starch$Ghat_perm),sd=result.starch$sd_eff),col="red",lwd=2,type="l",yaxt="n",ylab="",xlab="",xaxt="n",main="")
rect(par("usr")[1], par("usr")[3], par("usr")[2], par("usr")[4], col = "gray")
lines(seq(left,right,length.out=1000),dnorm(seq(left,right,length.out=1000),mean=mean(result.starch$Ghat_perm),sd=result.starch$sd_eff),col="red",lwd=2)
abline(v=result.starch$Ghat,lwd=2,col="darkblue")
text(par("usr")[1]+.18*(par("usr")[2]-par("usr")[1]),par("usr")[3]+0.8*(par("usr")[4]-par("usr")[3]),paste("p =  \n", round(result.starch$p.val,3)),cex=1.1,font=2)


#yield
par(fig=c(.83-.275-.5,.99-.275-.5,.85-.666666,.965-.666666),mgp=c(1,.3,0), new=TRUE)
par(mar=c(2,2,0,0))
left <- min(mean(result.yield$Ghat_perm)-4*result.yield$sd_eff,result.yield$Ghat)
right <- max(mean(result.yield$Ghat_perm)+4*result.yield$sd_eff,result.yield$Ghat)
plot(seq(left,right,length.out=1000),dnorm(seq(left,right,length.out=1000),mean=mean(result.yield$Ghat_perm),sd=result.yield$sd_eff),col="red",lwd=2,type="l",xaxt="n",yaxt="n",ylab="",xlab="")
rect(par("usr")[1], par("usr")[3], par("usr")[2], par("usr")[4], col = "gray")
lines(seq(left,right,length.out=1000),dnorm(seq(left,right,length.out=1000),mean=mean(result.yield$Ghat_perm),sd=result.yield$sd_eff),col="red",lwd=2)
abline(v=result.yield$Ghat,lwd=2,col="darkblue")
text(par("usr")[1]+.18*(par("usr")[2]-par("usr")[1]),par("usr")[3]+0.8*(par("usr")[4]-par("usr")[3]),paste("p =  \n", round(result.yield$p.val,3)),cex=1.1,font=2)

#drymat
par(fig=c(.83-.275,.98-.275,.85-.666666,.965-.666666),mgp=c(1,.3,0), new=TRUE)
par(mar=c(2,2,0,0))
left <- min(mean(result.drymat$Ghat_perm)-4*result.drymat$sd_eff,result.drymat$Ghat)
right <- max(mean(result.drymat$Ghat_perm)+4*result.drymat$sd_eff,result.drymat$Ghat)
plot(seq(left,right,length.out=1000),dnorm(seq(left,right,length.out=1000),mean=mean(result.drymat$Ghat_perm),sd=result.drymat$sd_eff),col="red",lwd=2,type="l",yaxt="n",ylab="",xaxt="n",xlab="")
rect(par("usr")[1], par("usr")[3], par("usr")[2], par("usr")[4], col = "gray")
lines(seq(left,right,length.out=1000),dnorm(seq(left,right,length.out=1000),mean=mean(result.drymat$Ghat_perm),sd=result.drymat$sd_eff),col="red",lwd=2)
abline(v=result.drymat$Ghat,lwd=2,col="darkblue")
text(par("usr")[1]+.18*(par("usr")[2]-par("usr")[1]),par("usr")[3]+0.8*(par("usr")[4]-par("usr")[3]),paste("p =  \n", round(result.drymat$p.val,3)),cex=1.1,font=2)




dev.off()

```


Now make pretty cumulative plots for each trait, like Jochen did with milk yield. SKIP FOR NOW

```{r, eval=F}
cumulative <- data.frame(adf=rep(NA,nrow(wqsMap)),ndf=NA,ivd=NA,cp=NA,starch=NA,yield=NA,drymat=NA)
for(i in 1:nrow(cumulative)){
  if(i==1){
  cumulative$adf[i] <- change[i]*effects.ADF[i]
  cumulative$ndf[i] <- change[i]*effects.NDF[i]
  cumulative$ivd[i] <- change[i]*effects.IVD[i]
  cumulative$cp[i] <- change[i]*effects.CP[i] 
  cumulative$starch[i] <- change[i]*effects.starch[i]
  cumulative$yield[i] <- change[i]*effects.yield[i]
  cumulative$drymat[i] <- change[i]*effects.drymat[i]     
  }
  if(i>1){
  cumulative$adf[i] <- cumulative$adf[i-1]+change[i]*effects.ADF[i]
  cumulative$ndf[i] <- cumulative$ndf[i-1]+change[i]*effects.NDF[i]
  cumulative$ivd[i] <- cumulative$ivd[i-1]+change[i]*effects.IVD[i]
  cumulative$cp[i] <- cumulative$cp[i-1]+change[i]*effects.CP[i] 
  cumulative$starch[i] <- cumulative$starch[i-1]+change[i]*effects.starch[i]
  cumulative$yield[i] <- cumulative$yield[i-1]+change[i]*effects.yield[i]
  cumulative$drymat[i] <- cumulative$drymat[i-1]+change[i]*effects.drymat[i]  
  }
}
### Everything below is depreciated.

### Plots with thresholds
#pdf("cumulativePlots_rrblup.pdf")
#plot(cumulative$adf,col=wqsMap$Chromosome, main="Cumulative ADF #freq*eff",xlab="Chromosome",xaxt="n",ylim=c(-1.5,1.5))
#axis(1,c(-10000,700,2000,3300,4300,5300,6300,7000,8000,8700,9700,100000),lab=c(NA,1:10,NA))
#abline(h=quantile(adf_test,c(0.025,0.975)),col="red")
#legend("topright","c(x,y)","Significance thresholds",col="red",pch="=")

#plot(cumulative$ndf,col=wqsMap$Chromosome, main="Cumulative NDF freq*eff",xlab="Chromosome",xaxt="n",ylim=c(-3,3))
#axis(1,c(-10000,700,2000,3300,4300,5300,6300,7000,8000,8700,9700,100000),lab=c(NA,1:10,NA))
#abline(h=quantile(ndf_test,c(0.025,0.975)),col="red")
#legend("topright","c(x,y)","Significance thresholds",col="red",pch="=")

#plot(cumulative$ivd,col=wqsMap$Chromosome, main="Cumulative IVTD freq*eff",xlab="Chromosome",xaxt="n",ylim=c(-1e-05,1e-05))
#axis(1,c(-10000,700,2000,3300,4300,5300,6300,7000,8000,8700,9700,100000),lab=c(NA,1:10,NA))
#abline(h=quantile(ivd_test,c(0.025,0.975)),col="red")
#legend("topright","c(x,y)","Significance thresholds",col="red",pch="=")
#
#plot(cumulative$cp,col=wqsMap$Chromosome, main="Cumulative CP freq*eff",xlab="Chromosome",xaxt="n",ylim=c(-1.5,1.5))
#axis(1,c(-10000,700,2000,3300,4300,5300,6300,7000,8000,8700,9700,100000),lab=c(NA,1:10,NA))
#abline(h=quantile(cp_test,c(0.025,0.975)),col="red")
#legend("topright","c(x,y)","Significance thresholds",col="red",pch="=")

#plot(cumulative$starch,col=wqsMap$Chromosome, main="Cumulative starch freq*eff",xlab="Chromosome",xaxt="n",ylim=c(-2,2))
#axis(1,c(-10000,700,2000,3300,4300,5300,6300,7000,8000,8700,9700,100000),lab=c(NA,1:10,NA))
#abline(h=quantile(starch_test,c(0.025,.975)),col="red")
#legend("topright","c(x,y)","Significance thresholds",col="red",pch="=")

#plot(cumulative$yield,col=wqsMap$Chromosome, main="Cumulative yield freq*eff",xlab="Chromosome",xaxt="n",ylim=c(-4e-06,4e-06))
#axis(1,c(-10000,700,2000,3300,4300,5300,6300,7000,8000,8700,9700,100000),lab=c(NA,1:10,NA))
#abline(h=quantile(yield_test,c(0.025,0.975)),col="red")
#legend("topright","c(x,y)","Significance thresholds",col="red",pch="=")

#plot(cumulative$drymat,col=wqsMap$Chromosome, main="Cumulative dry matter freq*eff",xlab="Chromosome",xaxt="n",ylim=c(-2,2))
#axis(1,c(-10000,700,2000,3300,4300,5300,6300,7000,8000,8700,9700,100000),lab=c(NA,1:10,NA))
#abline(h=quantile(drymat_test,c(0.025,0.975)),col="red")
#legend("topright","c(x,y)","Significance thresholds",col="red",pch="=")
#dev.off()
```